wandb/sdk/wandb_run.py

```diff
@@ -373,6 +377,9 @@ def __init__(
 
         # Initialize telemetry object
         self._telemetry_obj = telemetry.TelemetryRecord()
+        self._telemetry_obj_active = False
+        self._telemetry_obj_flushed = b""
+        self._telemetry_obj_updated = False
```

Comment:
do we need updated? can we just use the check that the `_telemetry_obj_flushed` is different from `_telemetry_obj`?

Reply:
If you compare then you pay the penalty of serializing the object so i was trying to avoid that.   in the current code it doesnt save us  much, but i thought it might come in handy more in the future.

Reply:
ah got ya...
nit: but also the name a bit confusing this flag is trying to say `need-to-update` but `updated` reads as `been-updated`

Reply:
i see.. it was meant to mean that _telemetry_obj has been updated... 
i can call it _telemetry_obj_dirty

Reply:
ah yeah... i think that's more clear

Reply:
or another option: 
`self._telemetry_obj_flushed` <- `self._telemetry_obj_latest` (this one might need something else...)
`self._telemetry_obj_updated` <- `self._telemetry_obj_flush`

Reply:
i changed it to dirty.
flushed binary, and the _obj is a protocol buf object, so i  wanted to keep those a bit differently named